### PR TITLE
Colors

### PR DESCRIPTION
**Pull request by gcstaublin**
## What this did

Creates variations and ranges for color palette.

2016-01-22 03 20 42 pm
2016-01-22 03 20 56 pm

2016-01-22 03 21 08 pm
2016-01-22 03 21 23 pm
2016-01-22 03 21 36 pm
2016-01-22 03 21 48 pm
2016-01-22 03 22 00 pm
2016-01-22 03 22 15 pm
2016-01-22 03 22 28 pm
2016-01-22 03 22 39 pm

## Why do this (discussion points)
Let's take our Dark Ui colors, for instance. We offer three shades within that palette, but that doesn't cover all our needs. Shot charts, as an example, use colors that 'of' the Dark UI palette but have been specifically created for the needs of the shot chart.

In other cases, the gradations of color give us the hover/active states we might need vs a button color being darkened by 20% here, but only by 15% over there. These variations also serve to offer options, depending on context. Let's say a blue button is on a dark background. In that case, the regular blue might be too contrasty, where as, the next shade or two darker, remains with that palette but is less abrupt. We'd want to provide the guidelines for usage, so it's universally applied but that's just a couple instances where these variations could come in handy.

Overall, these variations could stand to be tweaked. All values are 'nailed' It's a starting point.

Also, the Gray UI and Text UI palettes are pretty similar. We can narrow down to one palette that not only serves the Gray UI needs, as well as our Text color needs.

We'll also want to decide some interactions. Like when someone hovers on a button, what happens? Does the button darken? Does it lighten? Does it spin around three time and shoot off the canvas?

### CODE REVIEW

**Comment by gcstaublin** [comment]
^ discussion points.

**Comment by gcstaublin** [comment]
this one is ready for some eyes, too

**Comment by nick-adams** [comment]
* I agree with the merging of the gray palettes. Seems like we have some unnecessary distinction at the moment. Now, we could still create some use-case specific variable names that just reference the underlying gray colors, but at the very least we should consolidate on a spectrum of grays for any use.
* I wonder about some of the variations around the Hudl colors like orange and blue. I wonder if we want to keep those as a single color. Actually, I think we probably do since they are more closely tied to our brand (now, you could argue, why have a primary action color _be_ a brand color, but thats a separate argument that I lost once).
* I also wonder about the variations on the green, yellow, and red. If we were treating them as alert colors, it seems like we would want to remove this variation. Green, the same green, should always mean good/add/create everywhere, Red, the same red, should always mean bad/delete/ everywhere, etc. But I'm less sure of that one.

**Comment by gcstaublin** [comment]
I think we definitely want to create specific variables names that reference the building block variables. That's a philosophy we should get folks accustomed to following. If you use `$gray_color_light` in your stuff and then there a Kickoff change, or if you yourself want to update the usage, you're updating every instance of that variable, versus changing the reference of that variable.

My thought on the variations for colors like Orange or Green, etc. is to account for states. So hovering, or disabled, or what have you. In these cases, simply lightening or darkening a color might better solve the problem, but I do think there's value in having those use cases codified so people are not deciding on their own what happens when you hover on a confirm button.

**Comment by nick-adams** [comment]
Makes sense on those variations then. I think we can clarify that through documentation and the variable names themselves.

**Comment by chrisronline** [comment]
> If you use $gray_color_light in your stuff and then there a Kickoff change, or if you yourself want to update the usage, you're updating every instance of that variable, versus changing the reference of that variable.

Can you clarify what this means exactly @gcstaublin?

**Comment by gcstaublin** [comment]
Let me see if I can come up with a good example from within Kickoff itself. Let's say we have a default color for text within Hudl, and for all intents and purposes it's based on a variable named `$gray_color_light`. Rather, than referencing that variable directly in all the places within Kickoff that use that color, we'd want that abstracted into a variable call `$really_sweet_text_color`. And that `$really_sweet_text_color` would reference `$gray_color_light`. If it was decided that the default text color needed to change, we're changing `$really_sweet_text_color` and not all the instances of `$gray_color_light` that pertain to text color. Does that make better sense?

**Comment by chrisronline** [comment]
So, at the base level, you're defining variables based on the actual content - the color, but you also want to create aliases based on business logic - "primary" or "secondary". Is that right?

**Comment by gcstaublin** [comment]
2016-02-01 11 41 33 am

Here's an actual example. With the buttons, we reference those building block variables. Let's say, for whatever reason, the design system changes and the color for a Primary button in no longer Hudl Orange. Hudl orange is still a brand color used in key places, but for our primary button we want to change that. If we didn't have the `$btn_bg_primary` we'd hunt through Kickoff, looking for places where the color was used and figuring out if that applies to buttons or not. From experience, getting all those instances removed usually takes a few passes before you hit them all.

**Comment by nick-adams** [comment]
This reflects the reality that use-cases and business logic are not likely to change (primary will always be primary, etc.), but the underlying values that inform those use-cases are likely to change (what color orange we're using for primary).

By abstracting to that level, people will always use the use-case name in their styling, which means they are unlikely to ever have to refactor -- even if we change what hex value = orange_color 200 times. In that situation, we're just updating the underlying variable and it propagates automatically.

File: static/scss/variables/_colors.scss [hunk 1]
```diff
@@ -10,37 +10,101 @@
...
 // Dark UI Colors
 // --------------------------------------------
-$ui_dark_color:               #232A31;
-$ui_dark_medium_color:        #38434F;
-$ui_dark_light_color:         #4E5D6C;
+
+$ui_dark_100:                 #232a31;
+$ui_dark_80:                  #38434f;
+$ui_dark_60:                  #4E5D6C;
+$ui_dark_40:                  #7d90a2;
```
Comment (mfouquet) [inline static/scss/variables/_colors.scss]: Minor niggle that might be reserved for the "cleanup" pass, but can we be sure to get the capitalization consistent on these? My vote is for all caps. FIGHT

Comment (gcstaublin) [inline static/scss/variables/_colors.scss]: The should definitely be consistent.

Is `#9ba9b8` easier to read than `#9BA9B8`? With upper/lower does the numerical versus alpha values scan more quickly? Does that matter?

Comment (nick-adams) [inline static/scss/variables/_colors.scss]: Lowercase letters is a touch easier to read and requires one less keystroke. Thats the extent of the shits I give about it though.

**Pull request merged**